src/queue.rs

```diff
@@ -1,5 +1,19 @@
 // Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
 
+/// A cell containing a task and needed extra information.
+pub trait TaskCell {
+    /// Extra information in the cell.
+    type Extras;
+
+    /// Gets mutable extra information.
+    ///
+    /// # Safety
+    ///
+    /// This method is not safe. The caller must ensure that he is the only
+    /// one accessing the extras.
+    unsafe fn mut_extras(&mut self) -> &mut Self::Extras;
```

Comment:
I think it should be `&self` instead. Rust guarantees only one can access a `&mut` method.